misc/scripts/deploy-edgeX-snaps.sh

```diff
@@ -0,0 +1,169 @@
+  sudo snap install "$EF_SNAP_VAL" --dangerous
+elif [ $EF_SNAP_VAL_SET == true ] && [ $EF_SNAP_VAL_VALID == false ]; then
+  echo "$(timestamp) Local edgexfoundry snap not found or path invalid - $EF_SNAP_VAL"
+  echo "$(timestamp) Unable to continue with deployment"
+  set -o pipefail
+  exit
+elif [[ $EF_CHANNEL_VAL_SET == true ]]; then
+  echo "$(timestamp) Using edgexfoundry channel $EF_CHANNEL_VAL"
+  sudo snap install edgexfoundry "$EF_CHANNEL_VAL"
+else
+  echo "$(timestamp) Using stable version of edgexfoundry"
+  sudo snap install edgexfoundry --stable
+fi
+
+echo "$(timestamp) Installing application service configurable snap..."
+if [ $ASC_SNAP_VAL_SET == true ] && [ $ASC_SNAP_VAL_VALID == true ]; then
+  echo "$(timestamp) Using local edgex-app-service-configurable snap with dangerous flag"
+  sudo snap install "$ASC_SNAP_VAL" --dangerous
+elif [ $ASC_SNAP_VAL_SET == true ] && [ $ASC_SNAP_VAL_VALID == false ]; then
+  echo "$(timestamp) Local app-service-configurable snap not found or path invalid - $ASC_SNAP_VAL"
+  echo "$(timestamp) Unable to continue with deployment"
+  set -o pipefail
+  exit
+elif [[ $ASC_CHANNEL_VAL_SET == true ]]; then
+  echo "$(timestamp) Using edgex-app-service-configurable channel $ASC_CHANNEL_VAL"
+  sudo snap install edgex-app-service-configurable "$ASC_CHANNEL_VAL"
+else
+  echo "$(timestamp) Using stable version of edgex-app-service-configurable"
+  sudo snap install edgex-app-service-configurable
+fi
+
+echo "$(timestamp) Turning on sys-mgmt-agent service..."
+sudo snap set edgexfoundry sys-mgmt-agent=on
+
+echo "$(timestamp) Turning on support-scheduler service..."
+sudo snap set edgexfoundry support-scheduler=on
+
+echo "$(timestamp) Turning on support-notifications service..."
+sudo snap set edgexfoundry support-notifications=on
+
+echo "$(timestamp) Turning on device-virtual service..."
+sudo snap set edgexfoundry device-virtual=on
+
+echo "$(timestamp) Setting edgex-app-service-configurable profile to blackbox-tests..."
+sudo snap set edgex-app-service-configurable profile=blackbox-tests
+
+echo "$(timestamp) Starting edgex-app-service-configurable service..."
+sudo snap start edgex-app-service-configurable.app-service-configurable
+
+echo "$(timestamp) Generating security token. Token is used for tests when in secure mode..."
+TOKEN=$(echo "$TEST_USER_ACCESS_TOKEN" | sed 's/.*: \([^.]*\.[^.]*\.[^.]*\).*/\1/')
+export TOKEN
+echo "$(timestamp) Token was generated is located in the following environment variable: TOKEN"
+
+case $1 in
+--unset-env-var)
+	echo "$(timestamp) Unsetting TEST_ENV, EF_CHANNEL, ASC_CHANNEL, EF_SNAP, ASC_SNAP environment variables after deployment"
+	unset TEST_ENV
+	unset EF_CHANNEL
+	unset ASC_CHANNEL
+  unset EF_SNAP
+  unset ASC_SNAP
+	;;
+esac
+
+echo "============================================================================================"
+echo "              Completed the deployment of edgex snap test service for Blackbox"
```

Comment:
Note, if someone had specified `--unset-env-var`, then this output doesn't really apply...

Reply:
Removed this flag as its not needed.